File: Capítulo_9/Recursão/Sequência de Fibonacci/Fibonacci iterativo/Código em Python/fibonacciIterative.py
def fibonacci_iterative(n):
    if n < 1: return 0
    if n <= 2: return 1
    
    fib_minus_2 = 0
    fib_minus_1 = 1
    fib_n = n

    for i in range(2, n + 1):
        fib_n = fib_minus_1 + fib_minus_2
        fib_minus_2 = fib_minus_1
        fib_minus_1 = fib_n

    return fib_n

print(fibonacci_iterative(10)) # 55

File: Capítulo_9/Recursão/Sequência de Fibonacci/Fibonacci iterativo/fibonacciIterative.js
// Implementamos a função fibonacci de modo iterativo, assim:
function fibonacciIterative(n) {
    if (n < 1) return 0;
    if (n <= 2) return 1;
    let fibMinus2 = 0;
    let fibMinus1 = 1;
    let fibN = n;
    for (let i = 2; i <= n; i++) {
        fibN = fibMinus1 + fibMinus2;
        fibMinus2 = fibMinus1;
        fibMinus1 = fibN;
    }
    return fibN;
}
 console.log(fibonacciIterative(10)) // 55